sympy/matrices/matrices.py

```diff
@@ -1594,6 +1596,12 @@ def __pow__(self, num):
                     return self._matrix_pow_by_recursion(num)
             return self._matrix_pow_by_recursion(num)
         elif isinstance(num, (Expr, float)):
+            num = sympify(num)
+            if (num >= 0) == True and not num % 1 == 0:
+                z = self.jordan_form()[1]
```

Comment:
I thought that `num.is_Integer` (not `num.is_integer`) could be used when we know that `num` cannot be an int. Currently all floats are handled by the Jordan block method, and that could also be the case after they had been sympified to Floats (which satisfy the condition `isinstance(num, Expr)`). But it may be worth investigating if the recursion method could be used with integer valued floats like 50.0 (in which case the first condition would have to be modified accordingly).